packages/block-editor/src/components/rich-text/content.js

```diff
@@ -29,7 +30,11 @@ export const Content = ( { value, tagName: Tag, multiline, ...props } ) => {
 		value = `<${ MultilineTag }></${ MultilineTag }>`;
 	}
 
-	const content = <RawHTML>{ value }</RawHTML>;
+	const content = (
+		<RawHTML>
+			{ value instanceof RichTextData ? value.toHTMLString() : value }
```

Comment:
Tbh, this is only for something internal, because RichText.Content still has to support string values. Externally, you shouldn't have to do things like this?

Reply:
we maybe shouldn't? but we _are_, which is why I brought it up.
in this PR we have complicated the external interface by supporting both, so we've made it harder for consumers to do the right thing. the method also requires a change, but at least should make calling code more durable after the update and into the future.

it's not a blocker, just something I can see as causing more headaches in the future when we try and track down the places that _do_ and _do not_ continue to handle the legacy string value

Reply:
Ok, ok, to clarify things I rearrange the component and actually realised that for the string value, we need to continue using RawHTML (basically a div with `dangerouslySetHtml`), while for the rich text data value we can convert it straight to a React tree at some point and avoid serialising entirely.